test/bmc/bmc_api_test.rb

```diff
@@ -21,6 +21,48 @@ def setup
     authorize user, pass
   end
 
+  def test_api_throws_401_error_when_auth_is_not_provided
+    Proxy::BMC::IPMI.stubs(:providers_installed).returns(['freeipmi'])
+    Proxy::BMC::Plugin.settings.stubs(:bmc_default_provider).returns('freeipmi')
+    auth = Object.new
+    auth.stubs(:provided?).returns(false)
+    auth.stubs(:basic?).returns(true)
+    auth.stubs(:credentials).returns(nil)
```

Comment:
auth.basic was removed.  Although I am unsure of how to mock without stubbing?  Can you give me an example?  I am now using mock() instead of Object.new but thats all I changed.